### ISSUE TITLE
Create conversion script for v1.8 to v2.0

### ISSUE THREAD

**Issue by kkoopa** [nodejs/nan#360]
Much of the breaking changes in 2.0 should be rather easy to fix automatically. The trickiest would be the introduction of `Maybe` types. These could be converted straight away by always doing `ToLocalChecked` or `FromJust`, but I don't think it is such a good idea to let people skip through fixing that properly. Maybe just detect them and print a message in the conversion report, that this needs to be converted manually, same could be added as a comment in the processed source code.

Simply using regexes without actually parsing the C++ code should work well enough in most cases, and since everybody use git, it will be easy to go through the diff and apply chunks in patch mode, ignoring erroneous replacements.

The other way would be to use libclang to actually parse the C++ code and do conversion there. This could be made robustly, but I don't think it is worth the effort compared to the former option.

TL;DR: 2to3.py

**Comment by kkoopa** [comment]
For example, something like this should already cover a lot:
```js
s.replace(/(\W)args(\W)/, function (match, p1, p2, offset, string) {
  return [p1, info, p2].join();
});

s.replace('NanScope()', 'NanScope scope');

s.replace('NanEscapableScope()', 'NanEscapableScope scope');

s.replace('NanReturnValue', 'info.GetReturnValue().Set');

s.replace('NanReturnUndefined', 'return');

s.replace('NanReturnNull', 'info.GetReturnValue().SetNull');
```
